src/core.js

```diff
@@ -0,0 +1,131 @@
+const Auth = require('./auth');
+const Request = require('./request');
+
+class Core {
+  constructor() {
+    this._pkg = {
+      name: require('../package.json').name,
+      version: require('../package.json').version,
+    };
+  }
+
+  /**
+   * Creates a new session
+   * @param {object} params
+   * @returns {promise} client
+   */
+  connect(params) {
+    return new Promise((resolve) => {
+      if (typeof params.key === 'undefined') {
+        throw new Error(`Invalid Param: ${this._pkg.name} requires a "key".`)
+      }
+  
+      if (typeof params.key !== 'string') {
+        if (typeof params.key.kid === 'undefined' || typeof params.key.private === 'undefined' ) {
+          throw new Error(`Invalid Params: ${this._pkg.name} RSA auth requires a "key.kid" and a "key.private".`)
+        }
+      }
+
+      // properties
+      this._key = params.key;
+      this._protocol = params.protocol || `https`
+      this._host = params.host || 'consensys-mesh.ga';
```

Comment:
Might have to make this `consensys-mesh.com` in light of our recent DNS issue

Reply:
Wilco